deployer/src/lib.rs

```diff
@@ -1,4 +1,604 @@
+            runtime_manager,
+            dal,
+            task_router: TaskRouter::default(),
+            deployment_state_machine_channel,
+            runtime_start_channel,
+            config,
+        }
+    }
+
+    pub async fn start(self) -> Result<()> {
+        // The deployments which are in the `Running` state are considered runnable and they are started again. Running the
+        // deployments happens after their associated services' sandboxes are healthy and we start them.
+        let runnable_deployments = self.dal.running_deployments().await?;
+        info!(count = %runnable_deployments.len(), "enqueuing runnable deployments");
+        for existing_deployment in runnable_deployments {
+            // We want to restart the corresponding deployment service container.
+            let state = self
+                .dal
+                .service(&existing_deployment.service_id)
+                .await?
+                .state;
+
+            // Clean the previous docker container if any.
+            let runnable_deployment = RunnableDeployment {
+                deployment_id: existing_deployment.id,
+                service_name: existing_deployment.service_name,
+                service_id: existing_deployment.service_id,
+                tracing_context: Default::default(),
+                claim: None,
+                target_ip: state.target_ip(self.config.network_name.as_str()).ok(),
+                is_next: existing_deployment.is_next,
+            };
+            let image_name = state
+                .image()
+                .map_err(|err| Error::Internal(err.to_string()))?;
+
+            self.instate_service(
+                runnable_deployment,
+                GitMetadata::default(),
+                image_name.clone(),
+                existing_deployment.idle_minutes,
+                false,
+            )
+            .await?;
+        }
+
+        let mut server_builder = Server::builder()
+            .http2_keepalive_interval(Some(Duration::from_secs(60)))
+            .layer(JwtAuthenticationLayer::new(AuthPublicKey::new(
+                shuttle_proto::auth::client(&self.config.auth_uri)
+                    .await
+                    .expect("auth service should be reachable"),
+            )))
+            .layer(ExtractPropagationLayer);
+        let bind_address = self.config.bind_address;
+        let svc = DeployerServer::new(self);
+        let router = server_builder.add_service(svc);
+
+        router
+            .serve(bind_address)
+            .await
+            .expect("to serve on address");
+        Ok(())
+    }
+
+    // Ensures this service is created and the runtime loaded & started. Important to note that this method
+    // can be called when starting the deployer, to pick up from persistence the existing deployments and
+    // reinstate them if they are on the running code path, but also when deploying a brand new deployment,
+    // storing it in the persistence.
```

Comment:
Super nit: this should be doc comment
```suggestion
    /// Ensures this service is created and the runtime loaded & started. Important to note that this method
    /// can be called when starting the deployer, to pick up from persistence the existing deployments and
    /// reinstate them if they are on the running code path, but also when deploying a brand new deployment,
    /// storing it in the persistence.
```